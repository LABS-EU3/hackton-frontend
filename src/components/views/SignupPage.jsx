import React, { useEffect } from "react";
import { Redirect, useLocation } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import queryString from "query-string";

import image from "../../assets/Signup.png";
import { UserOnboarding } from "../templates";
import { socialAuthLoad, verifyEmail } from "../../store/user/actions";

const SignupPage = () => {
  let { search } = useLocation();
  const dispatch = useDispatch();
  const { token } = useSelector(state => state.currentUser);
  const { state } = useLocation();

  useEffect(() => {
    const { google, github, verified } = queryString.parse(search);
    if (google || github) {
      dispatch(socialAuthLoad());
    }
    if (verified) {
      dispatch(verifyEmail());
    }
  }, [search, dispatch]);

  if (token) {
    return <Redirect to={state?.from || '/dashboard'} />;
  }

  return (
    <UserOnboarding
      ctaText="Sign Up"
      imageType={image}
      imageText="Sign Up now!!"
      formHeader="Create an account"
      formParagraph="Join hackathons or organise one yourself."
    />
  );
};

export default SignupPage;
