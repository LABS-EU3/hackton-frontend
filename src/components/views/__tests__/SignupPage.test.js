import React from 'react'
import { Router, Route } from 'react-router';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react'
import SignupPage from '../SignupPage';
const history = createMemoryHistory()

const renderWithRouter = Component => render(
  <Router history={history}>
    <Route component={Component} />
  </Router>
)

describe("It should render signup page correctly", () => {
  
    it("renders the SignUp component correctly", () => {
      const render = () => renderWithRouter(SignupPage)
      // const { getByText } = render(render);
      // const formTitle = getByText(/Create an account/i);
      expect(render).toMatchSnapshot();
      // expect(formTitle).toBeInTheDocument();
    });
  });