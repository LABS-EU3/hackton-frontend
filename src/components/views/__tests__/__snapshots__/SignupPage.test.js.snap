// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render signup page correctly renders the SignUp component correctly 1`] = `[Function]`;
